lib/api/unittest/CBoostedTreeInferenceModelBuilderTest.cc

```diff
@@ -215,10 +216,15 @@ BOOST_AUTO_TEST_CASE(testIntegrationClassification) {
     BOOST_REQUIRE_EQUAL(expectedSize, trainedModel->size());
     BOOST_TEST_REQUIRE("logistic_regression" ==
                        trainedModel->aggregateOutput()->stringType());
+    const auto& classificationLabels{trainedModel->classificationLabels()};
+    BOOST_REQUIRE(classificationLabels.is_initialized() == true);
```

Comment:
I think it's setting a bad example that people will accidentally copy to use `BOOST_REQUIRE` with an assertion that has a simple comparison.  In this case it's obvious that if it fails the actual value was `false` instead of `true`, but if people copy the macro to other places where the RHS is not a boolean then we'll lose information for less obvious failures.

Therefore I think this line should be changed to either:

```
    BOOST_TEST_REQUIRE(classificationLabels.is_initialized() == true);
```

or:

```
    BOOST_REQUIRE(classificationLabels.is_initialized());
```

Reply:
Thank you for spotting this. I changed to the first option.